rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -100,6 +100,8 @@ rcl_logging_ret_t rcl_logging_external_initialize(
   const char * config_file,
   rcutils_allocator_t allocator)
 {
+  RCUTILS_CHECK_ALLOCATOR(&allocator, return RCL_LOGGING_RET_ERROR);
```

Comment:
Note that an "invalid argument" return code would be more appropriate, but this error code has already been established for at least some cases of an invalid allocator.

Reply:
the same check can be applied to `rcl_logging_get_logging_directory` as a function scope?

Reply:
> Note that an "invalid argument" return code would be more appropriate, but this error code has already been established for at least some cases of an invalid allocator.

Doing a quick grep for `RCUTILS_CHECK_ALLOCATOR` through our codebase, it looks like we most often use `INVALID_ARGUMENT` as the return value.  So I'd suggest that we switch to that here, as well as update the documentation for `rcl_logging_external_initialize` to say that it is a possible return code.

> the same check can be applied to `rcl_logging_get_logging_directory` as a function scope?

Yeah, I think that is a good idea.

Reply:
Just so we're clear: this will _technically_ be an API break as is demonstrated by the need to update the test.

Reply:
> Just so we're clear: this will _technically_ be an API break as is demonstrated by the need to update the test.

Yeah, I think that is OK here.  This is a really low-level function, and I don't really expect anything outside of `rcl` to be calling it.  Indeed, looking through all of the packages released into Rolling, nothing else calls it.

I also looked through some external rcl* packages:

* [ros2-rust](https://github.com/search?q=repo%3Aros2-rust%2Fros2_rust%20rcl_logging_external_initialize&type=code) - doesn't use it
* [ros2-java](https://github.com/search?q=repo%3Aros2-java%2Fros2_java%20rcl_logging_external_initialize&type=code) - doesn't use it
* [ada-ros](https://github.com/search?q=repo%3Aada-ros%2Frclada%20rcl_logging_external_initialize&type=code) - does use it, but since the repo hasn't been updated in 3 years I think there will be other breakage as well
* [r2r](https://github.com/search?q=repo%3Asequenceplanner%2Fr2r%20rcl_logging_external_initialize&type=code) - doesn't use it
* [rclc](https://github.com/search?q=repo%3Aros2%2Frclc%20rcl_logging_external_initialize&type=code) - doesn't use it
* [rclnodejs](https://github.com/search?q=repo%3ARobotWebTools%2Frclnodejs%20rcl_logging_external_initialize&type=code) - doesn't use it

I also did a [full search on GitHub](https://github.com/search?q=rcl_logging_external_initialize&type=code) for it, and while there are a few more uses of it, I don't think anything catastrophic is going to happen if we add this return code here.